jwst/master_background/expand_to_2d.py

```diff
@@ -265,13 +265,17 @@ def bkg_for_ifu_image(input, tab_wavelength, tab_background):
             x, y = grid_from_bounding_box(ifu_wcs.bounding_box)
             wl_array = ifu_wcs(x, y)[2]
             wl_array[np.isnan(wl_array)] = -1.
-            # flag values outside of the background wavelength table
+
+            # mask wavelengths not covered by the master background
             mask_limit = (wl_array > max_wave) | (wl_array < min_wave)
             wl_array[mask_limit] = -1
 
+            # mask_limit is indices into each WCS slice grid.  Need them in
+            # full frame coordinates, so we use x and y, which are full-frame
+            full_frame_ind = y[mask_limit].astype(int), x[mask_limit].astype(int)
+            background.dq[full_frame_ind] |= dqflags.pixel['DO_NOT_USE']
             # TODO - add another DQ Flag something like NO_BACKGROUND when we have space in dqflags
-            background.dq[mask_limit] = np.bitwise_or(background.dq[mask_limit],
-                                                      dqflags.pixel['DO_NOT_USE'])
```

Comment:
The bug was here @philhodge. `mask_limit` is a boolean array and needs to be the same size as the array it is indexing.  But it is the size of a slice bounding box, not the size of the full frame.

Reply:
Ok I am puzzled. When I put this code in I tested it with the MIRI MRS data. Both
IFUs have slices but in expand_to_2d.py the entire image is being sent - not just a slice.
At least that is the way I understood the code. Since this works fine for the MIRI MRS 
I don't understand why it does not work for the NIRSPEC IFU.

Reply:
NIRSpec IFU has multiple WCS object, one for each slice, each with its own bounding box.  I believe MIRI IFU is implemented differently, where there is a single WCS that gives the mapping for all the IFU slices, with NaNs in areas outside the slice bounding boxes.  Is this essentially the case @nden?

Reply:
Oh yes I forgot about how NIRSPEC is set up.

Reply:
Yes